packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -152,7 +148,9 @@ export class WaferMap extends FoundationElement {
                 return;
             }
             const { height, width } = entry.contentRect;
-            this.canvasSideLength = Math.min(height, width);
+            this.canvas.width = width;
```

Comment:
Should I add an additional step with an observable property and a changed callback or is it enough this way?

Reply:
I'm not sure I follow the question. Which property are you suggesting might be observable and have a changed callback?

Reply:
so before this change, the flow was resizeObserver->canvasSideLength->changed->queueRender
now it is just resizeObserver->queueRender because I change the canvas dimensions directly and I was wondering if I should change this to the initial flow

Reply:
Gotcha. The simpler flow seems fine to me. And I believe it still respects our goal of the render being triggered by component state changes and the actual rendering code only depending on state of the component.

But I welcome input from @rajsite too (he's OOO today so may not respond until Monday).

Reply:
From a model-view separation of concerns perspective it seems a bit off as-is. We are using the state on the view element as part of the "data model" of the state of the wafer map.

I'd recommend we make the separate observable canvasWidth and canvasHeight properties on the wafermap and instead of mutating the `canvas.width` and `canvas.height` imperatively here, we instead bind to it from the template:

```
<canvas class="wafer-map-canvas" ${ref('canvas')} width="${x => x.canvasWidth}" height="${x => x.canvasHeight}"></canvas>
```

And then other locations in the code would reference `wafermap.canvasHeight` and `wafermap.canvasWidth` instead of reaching into the view directly.

In that scenario the resizeObserver would not do a `queueRender` directly but instead the canvasWidthChanged and canvasHeightChanged methods would do it (and queueRender already debounces updates so shouldn't be called extra).